lib/install/management/appliance.go

```diff
@@ -491,17 +486,36 @@ func (d *Dispatcher) createAppliance(conf *config.VirtualContainerHostConfigSpec
 	}
 
 	var info *types.TaskInfo
-	// create appliance VM
-	if d.isVC && d.vchVapp != nil {
-		info, err = tasks.WaitForResult(d.op, func(ctx context.Context) (tasks.Task, error) {
-			return d.vchVapp.CreateChildVM(ctx, *spec, d.session.Host)
-		})
-	} else {
-		// if vapp is not created, fall back to create VM under default resource pool
-		info, err = tasks.WaitForResult(d.op, func(ctx context.Context) (tasks.Task, error) {
-			return d.session.VMFolder.CreateVM(ctx, *spec, d.vchPool, d.session.Host)
-		})
+
+	// Create the VCH inventory folder
+	vchFolder := d.session.VMFolder
+	if d.isVC {
+		folderPath := fmt.Sprintf("%s", path.Join(d.session.VMFolder.InventoryPath, conf.Name))
+
+		d.op.Info("Creating the VCH folder")
+		vchFolder, err = d.session.VMFolder.CreateFolder(d.op, spec.Name)
+		if err != nil {
+			if soap.IsSoapFault(err) {
+				switch soap.ToSoapFault(err).VimFault().(type) {
+				case types.DuplicateName:
+					return fmt.Errorf("a VM or folder already exists on the path for VCH folder (%s)", folderPath)
+				}
+			} else if soap.IsVimFault(err) {
+				switch soap.ToVimFault(err).(type) {
+				case *types.DuplicateName:
+					return fmt.Errorf("a VM or folder already exists on the path for VCH folder (%s)", folderPath)
+				}
+			}
+
+			return fmt.Errorf("unexpected err (%s): %s", folderPath, err)
```

Comment:
Can we make this message more user-friendly? (Even minor tweaks like replacing "err" with the word "error" might really help readability.)

Reply:
agreed -- I was going to address it in a PR around upgrade / configure..but we'll clean it up before this merge

Reply:
I've addressed this in #7690